### ISSUE TITLE
UTF8 is Not a Function, Connection issue

### ISSUE THREAD

**Issue by abnersajr** [sidorares/node-mysql2#585]
I have been using Ubuntu Gnome latest version. I have a fake kinesis + lambda running locally. When data reach one point of my code then start a mysql connection. However, this error message had been displayed for me:
```
Debug: internal, implementation, error 
    TypeError: utf8 is not a function
    at Function.from (native)
    at Function.from (native)
    at Object.exports.encode (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/parsers/string.js:32:19)
    at Packet.writeNullTerminatedString (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/packets/packet.js:762:26)
    at HandshakeResponse.serializeResponse (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/packets/handshake_response.js:98:10)
    at HandshakeResponse.toPacket (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/packets/handshake_response.js:146:16)
    at ClientHandshake.sendCredentials (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/commands/client_handshake.js:63:44)
    at ClientHandshake.handshakeInit (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/commands/client_handshake.js:142:10)
    at ClientHandshake.Command.execute (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/commands/command.js:40:20)
    at Connection.handlePacket (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/connection.js:500:28)
    at PacketParser.onPacket (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/connection.js:94:16)
    at PacketParser.executeStart (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/packet_parser.js:77:14)
    at Socket.<anonymous> (/home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/lib/connection.js:102:29)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)

```

**Comment by sidorares** [comment]
Hi! Looks very strange, as if `Buffer.from` expects callback and we passing 'utf8'
What's node version? Which version of node-mysql2 you are using? Can you check if `safe-buffer` is installed ( which version in /home/abner/projects/Tracking-API/local-server/.tmp/.dist/node_modules/mysql2/node_modules/package.json ?)

**Comment by YUFENGWANG** [comment]
same problem here... node v5.7.0

**Comment by furqan-razzaq** [comment]
Having exactly same problem here as well. @node 4.4.7.
`Error message: utf8 is not a function
Stack trace: TypeError: utf8 is not a function
    at Function.from (native)
    at Function.from (native)
    at Object.exports.encode (./node_modules/mysql2/lib/parsers/string.js:32:19)
    at Packet.writeNullTerminatedString (./node_modules/mysql2/lib/packets/packet.js:762:26)
    at HandshakeResponse.serializeResponse (./node_modules/mysql2/lib/packets/handshake_response.js:98:10)
    at HandshakeResponse.toPacket (./node_modules/mysql2/lib/packets/handshake_response.js:146:16)
    at ClientHandshake.sendCredentials (./node_modules/mysql2/lib/commands/client_handshake.js:63:44)
    at ClientHandshake.handshakeInit (./node_modules/mysql2/lib/commands/client_handshake.js:142:10)
    at ClientHandshake.Command.execute (./node_modules/mysql2/lib/commands/command.js:40:20)
    at PoolConnection.Connection.handlePacket (./node_modules/mysql2/lib/connection.js:500:28)
    at PacketParser.onPacket (./node_modules/mysql2/lib/connection.js:94:16)
    at PacketParser.executeStart (./node_modules/mysql2/lib/packet_parser.js:77:14)
    at Socket.<anonymous> (./node_modules/mysql2/lib/connection.js:102:29)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:153:18)
    at Socket.Readable.push (_stream_readable.js:111:10)
    at TCP.onread (net.js:536:20)`

**Comment by sidorares** [comment]
@abnersajr @YUFENGWANG @furqan-razzaq I'm pretty sure it's a side effect of #583
Need to add `var Buffer = require('safe-fuffer')` at the beginning of `lib/parsers/string.js` ( or just update node if you can ). I'll try to publish release with a fix today

**Comment by furqan-razzaq** [comment]
@sidorares Thanks for quick reply. Awaiting the fix. Reverting to previous version for now.

**Comment by sidorares** [comment]
@furqan-razzaq #380 explains reasoning for having `safe-buffer`

**Comment by sidorares** [comment]
@furqan-razzaq published as v1.3.4
